src/extra/gizmos/context.cpp

```diff
@@ -0,0 +1,138 @@
+#include "context.hpp"
+#include "../gfx/shards_types.hpp"
+#include <gfx/context.hpp>
+#include <gfx/window.hpp>
+#include <object_var_scope.hpp>
+#include <params.hpp>
+
+namespace shards {
+namespace Gizmos {
+using namespace gfx;
+struct HelperContextShard : public gfx::BaseConsumer {
+  static SHTypesInfo inputTypes() { return CoreInfo::AnyType; }
+  static SHTypesInfo outputTypes() { return CoreInfo::AnyType; }
+  static SHOptionalString help() { return SHCCSTR("Provides a context for rendering helpers"); }
+
+  PARAM_PARAMVAR(_view, "View",
+                 "The view used to render the helpers."
+                 "When drawing over a scene, the view should be the same",
+                 {Type::VariableOf(gfx::Types::View)});
+  PARAM_PARAMVAR(_queue, "Queue", "The queue to draw into", {Type::VariableOf(gfx::Types::DrawQueue)});
+  PARAM(ShardsVar, _content, "Content", "Content", {CoreInfo::ShardsOrNone});
+  PARAM_IMPL(HelperContextShard, PARAM_IMPL_FOR(_view), PARAM_IMPL_FOR(_queue), PARAM_IMPL_FOR(_content));
+
+  Context _context{};
+  SHVar *_contextVarRef{};
+
+  int2 _cursorPosition{};
+  bool _mouseButtonState{};
+
+  void warmup(SHContext *context) {
+    baseConsumerWarmup(context, true);
+
+    // Reference context variable
+    _contextVarRef = referenceVariable(context, Context::contextVarName);
+
+    {
+      ObjectVarScope _scope(*_contextVarRef, &_context, Context::Type);
+      PARAM_WARMUP(context);
+    }
+  }
+
+  void cleanup() {
+    {
+      ObjectVarScope _scope(*_contextVarRef, &_context, Context::Type);
```

Comment:
This is weird what is it?

Reply:
Replaced this scope thing with just a function with an inner callback.
I like that structure better for this.

Also fixed the scenario where they happen to be nested multiple levels